llvm/lib/Target/AArch64/AArch64ISelLowering.cpp

```diff
@@ -9455,9 +9574,14 @@ AArch64TargetLowering::LowerCall(CallLoweringInfo &CLI,
         DAG.getConstant(Intrinsic::aarch64_sme_set_tpidr2, DL, MVT::i32),
         DAG.getConstant(0, DL, MVT::i64));
     TPIDR2.Uses++;
+  } else if (RequiresSaveAllZA) {
+    Result = emitSMEStateSaveRestore(*this, DAG, FuncInfo, DL, Chain,
+                                     /*IsSave=*/false);
+    FuncInfo->setSMESaveBufferUsed();
```

Comment:
Would it make sense to push this into `emitSMEStateSaveRestore` for when `IsSave=false`? I ask because I'm assuming the "used" is synonymous with "has been restored".